Library/Homebrew/dev-cmd/pr-publish.rb

```diff
@@ -22,14 +24,13 @@ def pr_publish_args
   def pr_publish
     pr_publish_args.parse
 
-    ENV["HOMEBREW_FORCE_HOMEBREW_ON_LINUX"] = "1" unless OS.mac?
+    tap = Tap.fetch(Homebrew.args.tap || CoreTap.instance.name)
 
     args.named.uniq.each do |arg|
-      arg = "#{CoreTap.instance.default_remote}/pull/#{arg}" if arg.to_i.positive?
+      arg = "#{tap.default_remote}/pull/#{arg}" if arg.to_i.positive?
       url_match = arg.match HOMEBREW_PULL_OR_COMMIT_URL_REGEX
       _, user, repo, issue = *url_match
       odie "Not a GitHub pull request: #{arg}" unless issue
-      tap = Tap.fetch(user, repo) if repo.match?(HOMEBREW_OFFICIAL_REPO_PREFIXES_REGEX)
       ohai "Dispatching #{tap} pull request ##{issue}"
       GitHub.dispatch_event(user, repo, "Publish ##{issue}", pull_request: issue)
```

Comment:
This command takes a `tap` argument but dispatches against `repo`. I feel like this isn't quite correct but I'm not sure what the solution would be.

Reply:
I think it is good as it is now.